scico/solver.py

```diff
@@ -734,6 +734,7 @@ def __init__(
             W = snp.ones(A.shape[0], dtype=A.dtype)
         elif isinstance(W, Diagonal):
             W = W.diagonal
+            assert hasattr(W, "ndim")
```

Comment:
If this is to appease the typechecker, wouldn't something like
```
W: Array = W.diagonal
```
avoid the need for the assert?